tests/beaker_tests/vxlan_global/test_vxlan_global.rb

```diff
@@ -0,0 +1,222 @@
+# 0   - successful command execution
+# > 0 - failed command execution.
+#
+# Puppet Commands:
+# 0 - no changes have occurred
+# 1 - errors have occurred,
+# 2 - changes have occurred
+# 4 - failures have occurred and
+# 6 - changes and failures have occurred.
+#
+# NOTE: 0 is the default exit_code checked in Beaker::DSL::Helpers::on() method.
+#
+# The test cases use RegExp pattern matching on stdout or output IO
+# instance attributes to verify resource properties.
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# -----------------------------
+# Common settings and variables
+# -----------------------------
+testheader = 'Resource cisco_vxlan_global'
+
+# Define PUPPETMASTER_MANIFESTPATH.
+UtilityLib.set_manifest_path(master, self)
+
+# The 'tests' hash is used to define all of the test data values and expected
+# results. It is also used to pass optional flags to the test methods when
+# necessary.
+
+# 'tests' hash
+# Top-level keys set by caller:
+# tests[:master] - the master object
+# tests[:agent] - the agent object
+# tests[:show_cmd] - the common show command to use for test_show_run
+#
+tests = {
+  master: master,
+  agent:  agent,
+}
+
+# tests[id] keys set by caller and used by test_harness_common:
+#
+# tests[id] keys set by caller:
+# tests[id][:desc] - a string to use with logs & debugs
+# tests[id][:manifest] - the complete manifest, as used by test_harness_common
+# tests[id][:resource] - a hash of expected states, used by test_resource
+# tests[id][:resource_cmd] - 'puppet resource' command to use with test_resource
+# tests[id][:show_pattern] - array of regexp patterns to use with test_show_cmd
+# tests[id][:ensure] - (Optional) set to :present or :absent before calling
+# tests[id][:code] - (Optional) override the default exit code in some tests.
+#
+# These keys are local use only and not used by test_harness_common:
+#
+# tests[id][:manifest_props] - This is essentially a master list of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the list
+# tests[id][:resource_props] - This is essentially a master hash of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:title_pattern] - (Optional) defines the manifest title.
+#   Can be used with :af for mixed title/af testing. If mixing, :af values will
+#   be merged with title values and override any duplicates. If omitted,
+#   :title_pattern will be set to 'id'.
+#
+
+tests['preclean'] = {
+  manifest_props: '',
+  code:           [0, 2],
+  resource_props: {},
+}
+
+tests['default_properties'] = {
+  manifest_props: "
+    dup_host_ip_addr_detection_host_moves     => 'default',
+    dup_host_ip_addr_detection_timeout        => 'default',
+    anycast_gateway_mac                       => 'default',
```

Comment:
Does this mean that anycast_gateway_mac is not configured by default and is therefore not displayed in the `puppet resource` output?